packages/interbit-ui-components/src/components/ReduxForm/validation.js

```diff
@@ -0,0 +1,58 @@
+const email = value => {
+  const error = 'Invalid email address.'
+  if (
+    /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(value) ||
+    value === undefined ||
+    value === null ||
+    value === ''
+  ) {
+    return undefined
+  }
+  return error
+}
+
+const number = value => {
+  const error = 'Must be a number.'
+
+  if (
+    typeof value === 'object' ||
+    typeof value === 'boolean' ||
+    Number.isNaN(Number(value)) ||
+    !Number.isFinite(value)
+  ) {
+    return error
+  }
+  return undefined
+}
+
```

Comment:
There is something screwy about validating numbers. When you enter numbers into a form, the value being tested will be a string, not a number. What we care about for form validation, is that the value entered into the form can be interpreted as a number, which means that '100' is valid. I ended up removing number validation from the template for just this reason. If you accept my argument, then the validation logic can be simplified as:
```
// number will be undefined if value cannot be interpreted as a number
const number = Number(value)
if (Number.isFinite(number)) {
  return undefined
}
return error
```
(Bonus, supports Hex, Octal and binary numbers)

Reply:
This is much cleaner, but still allows boolean and `[]` as valid numbers. I've updated the validation function and tests so that it's clearer which values are valid, including "empty" ones, and which are invalid.